### PR TITLE
[FINERACT-1926] Loan account transfer POST API fixes

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Fix for initiating transfer on loan that is not in active status.
Additional validations.
Newly created owner id is saaved for transaction.

### CODE REVIEW

**Review by adamsaghy — APPROVED** [review]
LGTM

File: fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java [hunk 1]
```diff
@@ -117,7 +118,8 @@ private void validateSettlementDate(ExternalAssetOwnerTransfer externalAssetOwne
     }
 
     private void validateLoanStatus(LoanIdAndExternalIdData loanIdAndExternalIdAndExternalId) {
-        if (Objects.isNull(loanIdAndExternalIdAndExternalId)) {
+        if (Objects.isNull(loanIdAndExternalIdAndExternalId.getLoanId())
```
Comment (adamsaghy) [inline fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java:126]: Its a little bit ambigous... we will get the same error even the Loan is not even existing...

Comment (taskain7) [reply fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java:126]: fixed
it's got handled in the `LoanRepositoryWrapper`

File: fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java [hunk 2]
```diff
@@ -182,6 +189,25 @@ private void validateRequestBody(String apiRequestBodyAsJson) {
                         ExternalTransferRequestParameters.DATEFORMAT, ExternalTransferRequestParameters.LOCALE));
         final Type typeOfMap = new TypeToken<Map<String, Object>>() {}.getType();
         fromApiJsonHelper.checkForUnsupportedParameters(typeOfMap, apiRequestBodyAsJson, requestParameters);
+
+        final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+        final DataValidatorBuilder baseDataValidator = new DataValidatorBuilder(dataValidationErrors).resource("loantransfer");
+        final JsonElement json = fromApiJsonHelper.parse(apiRequestBodyAsJson);
+
+        String ownerExternalId = fromApiJsonHelper.extractStringNamed(ExternalTransferRequestParameters.OWNER_EXTERNAL_ID, json);
```
Comment (adamsaghy) [inline fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java:197]: This is required only for SALE.

File: fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java [hunk 3]
```diff
@@ -182,6 +189,25 @@ private void validateRequestBody(String apiRequestBodyAsJson) {
...
+        baseDataValidator.reset().parameter(ExternalTransferRequestParameters.OWNER_EXTERNAL_ID).value(ownerExternalId).notBlank()
+                .notExceedingLengthOf(100);
+
+        String transferExternalId = fromApiJsonHelper.extractStringNamed(ExternalTransferRequestParameters.TRANSFER_EXTERNAL_ID, json);
+        baseDataValidator.reset().parameter(ExternalTransferRequestParameters.TRANSFER_EXTERNAL_ID).value(transferExternalId).ignoreIfNull()
+                .notExceedingLengthOf(100);
+
+        String purchasePriceRatio = fromApiJsonHelper.extractStringNamed(ExternalTransferRequestParameters.PURCHASE_PRICE_RATIO, json);
```
Comment (adamsaghy) [inline fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java:205]: This is required only for SALE.

Comment (taskain7) [reply fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java:205]: based on the design doc, it's also in the request for buyback.

File: fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java (hunk 2 above)
Comment (taskain7) [reply fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java:197]: based on the design doc, it's also in the request for buyback.

**Pull request merged**